### PR TITLE
Remove trailing whitespace from all files

### PR DESCRIPTION
**Pull request by robbyoconnor**
Cosmetic change, strip trailing whitespace from all files.

### CODE REVIEW

**Comment by deb0ch** [comment]
Trailing whitespaces are dirty :poop:

**Comment by d12frosted** [comment]
Despite the fact that they are... well... white...

**Comment by robbyoconnor** [comment]
I actually had to script this -- sed is amazing.

File: core/core-funcs.el [hunk 1]
```diff
@@ -324,7 +324,7 @@ current window."
   (let ((comint-buffer-maximum-size 0))
     (comint-truncate-buffer)))
 
-
+
```
Comment (syl20bnr) [inline core/core-funcs.el]: Please update the script to keep the page breaks `^L` ;-)

Comment (syl20bnr) [inline core/core-funcs.el]: Please update the script to keep the page breaks ^L ;-)

Comment (robbyoconnor) [inline core/core-funcs.el]: Okay, will do.

**Comment by robbyoconnor** [comment]
@syl20bnr done -- this should be scripted -- and we REALLY should have a git hook to keep this from happening again.

File: core/tools/spacefmt/tests/test.org [hunk 2]
```diff
@@ -21,7 +21,7 @@
...
 * BAR
-** SUBBAR    ;     , .  
+** SUBBAR    ;     , .
```
Comment (robbyoconnor) [inline core/tools/spacefmt/tests/test.org]: LOL

**Comment by robbyoconnor** [comment]
I used this: 

```sh 
find . -type f \( -name "*.md" -o -name "*.org" -o -name "*.txt" -o -name "*.el" \)  -not -path ./elpa/* -not -path ./.cache/* -not -path ./.cask/*  -not -path ./.git/* -not -path ./session.* -exec perl -p -i -e "s/[ \t]$//g" {} \;
```

File: core/tools/spacefmt/tests/test.org (hunk 2 above)
Comment (syl20bnr) [inline core/tools/spacefmt/tests/test.org]: Is this whitespace in the test file on purpose ?

Comment (robbyoconnor) [inline core/tools/spacefmt/tests/test.org]: If it was -- I can fix it and adjust my command to exclude this file

Comment (robbyoconnor) [inline core/tools/spacefmt/tests/test.org]: I added it back.

**Comment by JAremko** [comment]
> we REALLY should have a git hook to keep this from happening again.

@robbyoconnor  But will a git-hook with `find` work on Windows? Also we probably can get a list of modified files(or even file fragments) and apply it to them.  This way we can avoid conflicts, when two PRs automatically try to remove the same whitespace. And it probably will be the best if the modification will be in the separate commit.

**Comment by robbyoconnor** [comment]
@JAremko -- Nope, it won't. That's just what I used -- if you look at it -- for every match it runs a perl script :-P

**Comment by robbyoconnor** [comment]
I'm not sure how it'd work on windows...We really need to spotcheck PRs.  Like we need to have people run `SPC x  d w` on their files when they make PRs.

**Comment by robbyoconnor** [comment]
I'm not sure how it'd work on windows...We really need to spotcheck PRs.  Like we need to have people run `SPC x  d w` on their files when they make PRs.  We COULD do it in fail if it finds any files using: 

```
find . -type f \( -name "*.md" -o -name "*.org" -o -name "*.txt" -o -name "*.el" \)  ! -path ./elpa/* ! -path ./.cache/* ! -path ./.cask/*  ! -path ./.git/* ! -path ./session.* ! -path core/tools/spacefmt/tests/test.org -exec perl -p -i -e "s/[ \t]$//g" {} \;
``` 

With a pre-commit hook, it would abort the commit operation if trailing whitespace exists. We can PROBABLY assume `sed` exists and use that.

**Comment by JAremko** [comment]
@robbyoconnor I thinkg I know how to do it :stuck_out_tongue: But it will be preatty insane. 

We can create a bot account for Spacemacs and add its credentials to the Travis build script that will be ran only when someone with write privilege pushes changes to the develop branch (only those accounts have accsess to the secure varable that will hold bot's  credentials)
The script will run `spacefmt`  and other tools. Then it will run git status and if there are any changes it will add extra commit with formatting and stuff, or open a PR (for example, if  there is a conflict)

I was thinknig about making something like this to autmaticaly open PRs for spacemacs.org  when the documentation updates.


*I really need to finish this spacemacs-docker thingy ASAP. And start doing something useful :smile: *

**Comment by robbyoconnor** [comment]
@JAremko -- You need to run this on all text, emacs lisp, org, and markdown files as well. People introduce trailing whitespace in all of those.

**Comment by JAremko** [comment]
@robbyoconnor  sure. We need to expand `spacefmt`. And fix #7675

**Comment by deb0ch** [comment]
@JAremko I think the idea of a githook is good, but IMHO it should just reject the push instead of adding a commit or a PR by itself. It doesn't sound very safe.

**Comment by robbyoconnor** [comment]
the git hook should fix the trailing whitespace, but not generate a commit.

**Comment by JAremko** [comment]
@robbyoconnor  @deb0ch  It can be in a hook but it will may be hard to make cross platform. Also it may be confusing with some github clients or the hook can be disabled. We can check it in a Travis script as a simple test(This way we don't need to care about compatibility) or make it produce PR with fixes - the biggest benefit of it will be that it will reduce the friction for contributors.

**Comment by JAremko** [comment]
>  We can PROBABLY assume sed exists and use that. 

@robbyoconnor    It's is [gsed](https://github.com/syl20bnr/spacemacs/blob/master/core/tools/spacefmt/spacefmt#L24)  on MacOS and I doubt it exist on Windows in any shape or form :disappointed:

**Comment by JAremko** [comment]
Actually, since we already have [the "wrong branch"](https://github.com/syl20bnr/spacemacs/blob/master/.travis-build.sh#L26) check in Travis, it may be even better if we extand it and insted of warning a contributer in the travis build log (that some people apparently don't read)  the bot will add a message to the PR thread with the list of the issues that have to be solved. Or even solve them and open a PR to the PR source fork with the fixes and then the contributor can review them and merge.

**Comment by robbyoconnor** [comment]
@JAremko whatever solution we come up with...not having to do this again would be great :)

**Comment by robbyoconnor** [comment]
@syl20bnr changes have been made :) I intentionally excluded @JAremko's test file. Let's think about an automated way of doing this. Trailing whitespace offends me greatly.

**Comment by robbyoconnor** [comment]
Some more changes -- When merging PRs -- let's check for trailing whitespace folks! 

@d12frosted, @syl20bnr

**Comment by JAremko** [comment]
@robbyoconnor  ok..  Doing it next :shipit:

**Comment by robbyoconnor** [comment]
@JAremko -- I excluded your test org file :)

**Comment by syl20bnr** [comment]
> I even created an account for the bot to honor the memories of @StreakyCobra https://github.com/robocobra

😂  I love the idea!

cc @StreakyCobra

**Comment by syl20bnr** [comment]
Thank you ! 💜 
Very nice cleanup, I have some work to do in my house too if you are interested 😆 

Cherry-picked into develop branch, you can safely delete your branch.

**Comment by robbyoconnor** [comment]
@syl20bnr -- unless that work can be scripted like this, nah. :)

**Comment by StreakyCobra** [comment]
> I even created an account for the bot to honor the memories of @StreakyCobra https://github.com/robocobra

:joy: 

> But then it went rogue and started trolling my comrades

I'm not really surprized :thinking:

**Pull request closed without merging**